test/lib/hakkasan.test.ts

```diff
@@ -0,0 +1,41 @@
+import { generateMockFetch } from 'test/testUtils';
+import {
+  parseHakkasanData,
+  fetchHakkasanByRef,
+  VALID_HAKKSAN_REFS,
+} from 'lib/hakkasan';
+import events from 'data/events.json';
+
+describe('parseHakkasanData(text)', () => {
+  test('should parse the argument of retrieveJSONP from Hakkasan JSON data', () => {
+    const data = { message: 'Test data' };
+    const dataText = JSON.stringify(data);
+    const text = `retrieveJSONP(${dataText})`;
+    const result = parseHakkasanData(text);
+    expect(result).toEqual(data);
+  });
+
+  test('should catch and rethrow errors', () => {
+    expect(parseHakkasanData).toThrowError();
+  });
+});
+
+describe('fetchHakkasanByRef(ref)', () => {
+  test('should not fetch Hakkasan data with an invalid ref', async () => {
```

Comment:
Hmm I'm thinking we might be able to remove this test case entirely by simply changing the typing of the `ref` argument of the function to be more accurately a string union of either `artists`, `events`, or `venues`.

With that we'd also be able to remove the `VALID_HAKKSAN_REFS` constant.

Reply:
Ooh yes, that sounds like a cleaner approach!